exonum/src/events/network.rs

```diff
@@ -215,7 +215,7 @@ impl ConnectionsPool {
         if let Some(remote_public_key) = connect_list.find_key_by_address(&peer) {
             let mut handshake_params = handshake_params.clone();
             handshake_params.set_remote_key(remote_public_key);
-            NoiseHandshake::initiator(&handshake_params).send(stream)
+            NoiseHandshake::initiator(&handshake_params, peer).send(stream)
```

Comment:
Maybe we could put peer address into `handshake_params` to avoid changing function signature. But TBH this socket address has nothing to do with Noise handshake so I have some doubts about that.

Reply:
I don't see problems in changing the function signature. You can add a comment why this parameter is needed if it bothers you.